crates/ruff_cli/src/commands/format.rs

```diff
@@ -567,3 +572,69 @@ impl Display for FormatCommandError {
         }
     }
 }
+
+pub(super) fn warn_incompatible_formatter_settings(
+    pyproject_config: &PyprojectConfig,
+    resolver: Option<&Resolver>,
+) {
+    for setting in std::iter::once(&pyproject_config.settings)
+        .chain(resolver.iter().flat_map(|resolver| resolver.settings()))
+    {
+        let mut incompatible_rules = Vec::new();
+
+        for incompatible_rule in RuleTable::from_iter([
+            Rule::LineTooLong,
+            Rule::TabIndentation,
+            Rule::IndentationWithInvalidMultiple,
+            Rule::IndentationWithInvalidMultipleComment,
+            Rule::OverIndented,
+            Rule::IndentWithSpaces,
+            Rule::SingleLineImplicitStringConcatenation,
+            Rule::MissingTrailingComma,
+            Rule::ProhibitedTrailingComma,
+            Rule::BadQuotesInlineString,
+            Rule::BadQuotesMultilineString,
+            Rule::BadQuotesDocstring,
+            Rule::AvoidableEscapedQuote,
```

Comment:
One tricky thing here is that these don't necessarily _conflict_ with the formatter. Some of them do _overlap_, but in a way that's at least consistent. Some of them are also _arguably_ complimentary (like `MissingTrailingComma`, I think, is arguably complimentary, since it will insert trailing commas that the formatter will then respect).

Reply:
`COM812` is conflicting because it disallows:

```python
class PhotoMetadata:
    """Metadata about a photo."""

    def __init__(
        self, file: Path, b, c, d, e, f, g, h, i, j, k, l, m, n, o, p, q, r, s, t, u
    ):
        ...
```

My goal is to only warn about rules that are incompatible. E.g. I excluded 

* E501 because there are valid use cases to use it alongside the formatter, even though it's mostly redudnant
* `E701`: while redundant, it isn't conflicting with the formatter.

Is your feedback mainly about the naming of the variables or the wording of the warning message?

Reply:
I think I'm just trying to understand what we consider to be a conflict (and, based on that answer, perhaps looking to refine the list, or change the wording of the message).

For example, above, if you ran the linter, wouldn't `COM812` just insert a trailing comma at the end, and then the formatter would in turn split the arguments over multiple lines? So it's not like the two tools would go back and forth endlessly disagreeing with one another, as they would if you set `BadQuotes` to divergent values. The `COM812` case feels like it fits some definition of a conflict though, like: formatted code can introduce these violations.

Reply:
> For example, above, if you ran the linter, wouldn't COM812 just insert a trailing comma at the end, and then the formatter would in turn split the arguments over multiple lines?

That's correct. But it requires that you need to run `formatter`, `linter`, and `formatter` again to get a stable result which doesn't seem ideal. I see incompatibilities as everything that leads to a bad experience.

Reply:
So, anything that might trigger or require some command to be repeated, even if that cycle isn't infinite. In that case, we probably want to include ISC001?

Reply:
Probably we should. This rule won't be conflicting with the preview style but the formatter could produce code today that conflicts with it